src/qililab/platform/platform.py

```diff
@@ -477,6 +477,9 @@ def compile(self, program: PulseSchedule | Circuit, num_avg: int, repetition_dur
         for pulse_bus_schedule in pulse_schedule.elements:
             bus = self.buses.get(port=pulse_bus_schedule.port)
             bus_programs = bus.compile(pulse_bus_schedule, num_avg, repetition_duration, num_bins)
-            programs[bus.alias] = bus_programs
+            # ensuring that buses without AWG type instruments but with non AWG instruments don't get added to the
+            # programs queue
+            if bus_programs:
+                programs[bus.alias] = bus_programs
```

Comment:
Wait, why is there a pulse_bus_schedule created for the bus without an awg??

Reply:
The problem might be somewhere else then!

Reply:
The fix returns None if there is no AWG present, but if we don't check in here it will create an entry in the dictionary with an empty list, I wanted to avoid that.

Reply:
If that is no problem, then I can just undo this one.

Reply:
Here is a small summary of what I think is happening:

For the Bluefors setup, we use QbloxQCM instruments to apply offsets to all the qubits. QbloxQCMs are not supposed to do this, that is why we have voltage/current sources (don't ask me why are we using them for this purpose, that's a HW decision 😆 ). 

Whatever, a big problem we saw is that the offsets of the QbloxQCMs are only activated when you are actually running a program (thus they cannot be used as constant voltage/current sources).

To solve this, we decided to create "empty programs", which are Qblox programs that don't do anything, they just have a wait time equal to all the other programs that are being executed in the other sequencers. This way, we can activate the offsets while running pulse sequences.

What I believe is happening here is that we are still creating this empty programs, and we are trying to send them to the SPIRack, which of course complains, because it is not an AWG.

Reply:
There are two places where this happens and fails, compile level and run level. I have removed the raised error in both, run is fine just by doing return if no AWG instrument is present, as the return type is not used anywhere. However compiling does something with the returned compiled program, which right now is None after the fix, but it could be an empty list. Whatever the return type is, it will get added to the dictionary of programs per bus, so I think we should avoid that.

Reply:
So, long story short, I wouldn't remove the error from the bus! What this error is saying (or should say) is that `Bus.compile` cannot be called when the bus doesn't contain an AWG. 

The actual error comes from before, during the transpilation:

https://github.com/qilimanjaro-tech/qililab/blob/0d2eb761fbde9e2bfb96cc4c61dbe614a803e3b9/src/qililab/pulse/circuit_to_pulses.py#L124-L129

This 5 lines create the empty pulse schedules that I was talking about. What I would do is add an `if` statement that only runs these lines when the flux buses are connected to AWGs.

Reply:
Obviously we should avoid to create empty programs, but not sure this hotfix is meant to tackle that bigger task, as I am not sure the implications of removing them, as you mentioned they were created as a way to by-pass certain HW limitations.

Reply:
We can discuss it in more detail after the backlog refinement meeting!

Reply:
> Obviously we should avoid to create empty programs, but not sure this hotfix is meant to tackle that bigger task, as I am not sure the implications of removing them, as you mentioned they were created as a way to by-pass certain HW limitations.

It is not a bigger task! It should be an extra 3 lines 😄 

It is very important that you understand the problem we are having here! Otherwise it would be impossible to fix it. The real problem is that we are creating empty schedules when we shouldn't be creating them (because of everything I mentioned before).

Reply:
Understood after meeting to understand the whole issue better with empty schedules.